### PR TITLE
MSI/ClientCert support in SignalR service

### PR DESCRIPTION
**Pull request by juniwang**
- Adding MSI support in SignalR Service
- Allow customer to enable client certificate for SignalR data plane auth.

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Contribution checklist:
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Service team MUST add the "**WaitForARMFeedback**" label if the management plane API changes fall into one of the below categories. 
  - adding/removing APIs.
  - adding/removing properties.
  - adding/removing API-version. 
  - adding a new service in Azure.

<i>Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.</i>

- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json [hunk 1]
```diff
@@ -1440,6 +1452,67 @@
...
+          "description": "The Resource indicating the App ID URI of the target resource.\r\nIt also appears in the aud (audience) claim of the issued token.",
+          "type": "string"
+        }
+      }
+    },
+    "ManagedIdentityParameters": {
+      "description": "A class represent managed identities used for request and response",
+      "type": "object",
+      "properties": {
+        "type": {
```
Comment (yungezz) [inline specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json:1495]: suggest to use enum as type, to align with other services, such as https://github.com/Azure/azure-rest-api-specs/blob/master/specification/web/resource-manager/Microsoft.Web/stable/2019-08-01/CommonDefinitions.json#L1228-L1275

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json [hunk 2]
```diff
@@ -1440,6 +1452,67 @@
...
+            "$ref": "#/definitions/UserAssignedIdentityProperty"
+          }
+        },
+        "principalId": {
+          "description": "Get or set the principal id for the system assigned identity.\r\nOnly be used in response.",
+          "type": "string"
+        },
+        "tenantId": {
+          "description": "Get or set the tenant id for the system assigned identity.\r\nOnly be used in response",
+          "type": "string"
```
Comment (yungezz) [inline specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json:1512]: is this readonly?

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json [hunk 3]
```diff
@@ -1440,6 +1452,67 @@
         }
       }
     },
+    "UpstreamAuthSettings": {
```
Comment (yungezz) [inline specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json:1455]: just want to understand the context, seems signalr has extra layer of model `UpstreamAuthSettings` ontop of `ManagedIdentity` settings, which is different with other services,  seems it's for extensibility? but currently only managedIdentity and none supported.

**Comment by ravbhatnagar** [comment]
Adding new properties requires a new api-version. Please get sign off/exception from Azure API review board (azureapirbcore) if you think these dont need a new api-version.

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json (hunk 3 above)
Comment (zackliu) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json:1455]: SignalR Service is a fully managed service, one of the functionality is the service can invokes other service using managed identity to acquire AAD token if customer choose to use `ManagedIdentity` in `UpstreamAuthSettings`. At the meanwhile, customer can specify the `target resource`. 
In simple, `ManagedIdentity` is to assign a identity just like other services. And `UpstreamAuthSettings` is to tell the service when to use managed identity to acquire aad token. `UpstreamAuthSettings` in each `upstream` can vary.

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json (hunk 2 above)
Comment (zackliu) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json:1512]: Yes

Comment (yungezz) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json:1512]: pls add readonly property then, same for other readyonly properties.

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json (hunk 1 above)
Comment (juniwang) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json:1495]: Updated to `enum`

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json (hunk 2 above)
Comment (juniwang) [reply specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json]: `readnly` added

**Comment by juniwang** [comment]
We believe it's not a breaking change and backward compatibility is carefully took care of.

**Comment by ravbhatnagar** [comment]
Removing ARM feedback label since ARM will not be reviewing the property level changes going forward.

File: specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json [hunk 4]
```diff
@@ -1440,6 +1452,82 @@
...
+      "description": "Managed identity settings for upstream.",
+      "type": "object",
+      "properties": {
+        "resource": {
+          "description": "The Resource indicating the App ID URI of the target resource.\r\nIt also appears in the aud (audience) claim of the issued token.",
+          "type": "string"
+        }
+      }
+    },
+    "ManagedIdentityParameters": {
```
Comment (johanste) [inline specification/signalr/resource-manager/Microsoft.SignalRService/stable/2020-05-01/signalr.json:1491]: Why the `Parameters` suffix on this model type?

**Comment by yungezz** [comment]
> Removing ARM feedback label since ARM will not be reviewing the property level changes going forward.

hi @ravbhatnagar , could you pls share more about the change: ARM will not reviewing property level changes going forward? Since this will impact our existing Swagger review process/tools etc. cc @akning-ms

**Pull request closed without merging**